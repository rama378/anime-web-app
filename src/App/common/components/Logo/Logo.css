.Logo {}

.Logo__icon {
  position: relative;
  display: inline-block;
  vertical-align: middle;
  background-color: var(--text-color);
  color: var(--background);
  width: 70px;
  height: 70px;
  border-radius: 16px;
}

.Logo__icontext {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-weight: bold;
  font-size: 20px;
}

.Logo__icontextcenter {
  font-size: 24px;
  vertical-align: middle;
}

.Logo__appname {
  display: inline-block;
  vertical-align: middle;
  font-weight: bold;
  margin: 0 0 0 12px;
  font-size: 20px;
}

@media only screen and (max-width: 500px) {
  .Logo__icon {
    width: 50px;
    height: 50px;
  }

  
  .Logo__icontext {
    font-size: 18px;
  }

  .Logo__icontextcenter {
    font-size: 22px;
  }

  .Logo__appname {
    font-size: 18px;
  }
}