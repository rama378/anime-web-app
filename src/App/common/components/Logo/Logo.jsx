import './Logo.css';

const Logo = () => {
	return (
		<div>
			<div className='Logo__icon'>
				<div className='Logo__icontext'>
					A<span className='Logo__icontextcenter'>W</span>A
				</div>
			</div>
			<div className='Logo__appname'>
				Anime Web App
			</div>
		</div>
	);
};

export default Logo;
