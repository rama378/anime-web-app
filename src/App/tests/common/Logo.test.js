import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Logo from '../../common/components/Logo/Logo';

const renderer = new ShallowRenderer();

describe('<Logo />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<Logo />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
