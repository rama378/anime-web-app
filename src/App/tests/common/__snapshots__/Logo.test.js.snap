// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render and match the snapshot 1`] = `
<div>
  <div
    className="Logo__icon"
  >
    <div
      className="Logo__icontext"
    >
      A
      <span
        className="Logo__icontextcenter"
      >
        W
      </span>
      A
    </div>
  </div>
  <div
    className="Logo__appname"
  >
    Anime Web App
  </div>
</div>
`;
